compiler/luci/import/src/CircleMetadataImporter.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CircleMetadataImporter.h"
+
+#include <set>
+#include <vector>
+
+namespace
+{
+
+uint32_t read_u32(const std::vector<uint8_t> &buffer, uint32_t idx)
+{
+  uint32_t val = 0;
+  val += (buffer.at(idx + 0) << 0 * 8);
+  val += (buffer.at(idx + 1) << 1 * 8);
+  val += (buffer.at(idx + 2) << 2 * 8);
+  val += (buffer.at(idx + 3) << 3 * 8);
+  return val;
+}
+
+} // namespace
+
+namespace
+{
+
+/**
+ * 'source_table' consists of following structure.
+ *  - [ entry_number : uint32_t ]
+ *  - [ id : uint32_t ][ length : uint32_t ][ data : 'length' Bytes ] * entry_number
```

Comment:
instead of `Bytes`, how about `uint8_t` as there is `uint32_t`

Reply:
um... like `[ data : uint8_t * length]` to follow `] * entry_number` part

Reply:
Then how about `[ data : char * length ]` ?
The data consists of `char` :D